app/api/routes.py

```diff
@@ -53,6 +53,9 @@ def downvote(id):
 def update_resource_click(id):
     return add_click(id)
 
+@bp.route('/resources/search/<string:term>', methods=['GET'])
```

Comment:
Do we want this to be just '/search/ '

Reply:
When you do a Google search, it puts your query in a query param called `q`, so the URL ends up something like `www.google.com/search?q=operation+code`. I think it would be good to follow this pattern, especially because I'm not sure how well path parameters will be able to be url encoded (i.e., replacing a space with `+`). Also, the search should be searching all the data (like languages and categories). So I think we should just make it `/search` instead of `/resources/search` and then look in the query params for `q` to find the specific search terms

Reply:
I agree with replacing the path parameters with regular params and changing it to /search.

My confusion is, when you say you want them to be able to search languages/categories, are you saying you want it to return a Language/Category object? If that's what you mean, could you explain the use case for that?

Otherwise, language/category is already being used within the search. We can add filters for categories/languages though.

Reply:
I just meant that if someone searches for `python` they're searching for a certain language, so calling it `/resources/search` isn't entirely correct. Although I could see the case for how you only want resources, because like you said, this endpoint will only be returning resources. That being said, it doesn't really matter, you already said you agree with changing it to `/search` which was the point of me saying that, so we're good.